src/auth-service/utils/request.js

```diff
@@ -54,105 +44,111 @@ const request = {
-      filterRequestBody["query"]["category"] = category;
+      let filter = { email };
 
-      const responseFromGenerateCandidateFilter =
-        generateFilter.candidates(filterRequestBody);
+      const responseFromListCandidates = await CandidateModel(
+        tenant.toLowerCase()
+      ).list({
+        filter,
+      });
 
-      if (responseFromGenerateCandidateFilter.success === false) {
-        callback({
-          success: false,
-          message: "internal server error",
-          errors: { message: responseFromGenerateCandidateFilter.error },
-        });
-      }
+      logObject("responseFromListCandidates", responseFromListCandidates);
 
-      let filter = responseFromGenerateCandidateFilter.data;
-      const responseFromListCandidates = await request.list({
-        tenant,
+      const responseFromListUser = await UserModel(tenant.toLowerCase()).list({
         filter,
       });
 
-      if (responseFromListCandidates.success === true) {
-        if (
-          Array.isArray(responseFromListCandidates.data) &&
-          responseFromListCandidates.data.length === 1
-        ) {
-          callback({
-            success: true,
-            message: "candidate already exists or successfully created",
-            status: httpStatus.OK,
-          });
-        } else {
-          const responseFromCreateCandidate = await CandidateModel(
-            tenant
-          ).register(req);
+      logObject("responseFromListUser", responseFromListUser);
 
-          if (responseFromCreateCandidate.success === true) {
-            let createdCandidate = await responseFromCreateCandidate.data;
-            let responseFromSendEmail = await mailer.candidate(
-              firstName,
-              lastName,
-              email,
-              tenant
-            );
-            if (responseFromSendEmail.success === true) {
-              const status = responseFromSendEmail.status
-                ? responseFromSendEmail.status
-                : "";
-              callback({
-                success: true,
-                message: "candidate already exists or successfully created",
-                data: createdCandidate,
-                status,
-              });
-            } else if (responseFromSendEmail.success === false) {
-              const errors = responseFromSendEmail.error
-                ? responseFromSendEmail.error
-                : "";
-              const status = responseFromSendEmail.status
-                ? responseFromSendEmail.status
-                : "";
+      if (
+        responseFromListCandidates.message ===
+          "successfully listed the candidates" ||
+        responseFromListUser.message === "successfully listed the users"
+      ) {
+        callback({
+          success: true,
+          message: "candidate or user already exists",
+          status: httpStatus.OK,
+        });
+      } else if (
+        responseFromListCandidates.message === "no users exist" ||
```

Comment:
this is where we are still resolving issues. @wabinyai

Reply:
@wabinyai , issue has been addressed. Let me leave some review comments to shed more light.
Happy reviewing!